docs/auth/vmware-cloud/provider.md

```diff
@@ -0,0 +1,250 @@
+---
+id: provider
+title: VMware Cloud Authentication Provider
+sidebar_label: VMware Cloud
+description: Adding VMware Cloud as an authentication provider in Backstage
+---
+
+Backstage comes with an auth provider module to allow users to sign-in with
+their VMware Cloud account. This page describes some actions within the VMware
+Cloud Console and within a Backstage app required to enable this capability.
+
+## Create an OAuth App in the VMware Cloud Console
+
+1. Log in to the [VMware Cloud Console](https://console.cloud.vmware.com).
+1. Navigate to [Identity & Access Management > OAuth
+   Apps](https://console.cloud.vmware.com/csp/gateway/portal/#/consumer/usermgmt/oauth-apps)
+   and click the [Owned
+   Apps](https://console.cloud.vmware.com/csp/gateway/portal/#/consumer/usermgmt/oauth-apps/owned-apps/view)
+   tab -- if you are not an Organization Owner or Administrator but only a
+   Member, you will not see this nav entry unless the **Developer** check box is
+   selected for your role (see the [Organization roles and
+   permissions](https://docs.vmware.com/en/VMware-Cloud-services/services/Using-VMware-Cloud-Services/GUID-C11D3AAC-267C-4F16-A0E3-3EDF286EBE53.html#organization-roles-and-permissions-0)
+   docs for details).
+1. Click **Create App**, choose 'Web/Mobile app' and click **Continue**.
+1. Use default settings except:
+   - `App Name` and `App Description` of your choosing.
+   - `Redirect URIs`: `${baseUrl}/api/auth/vmwareCloudServices/handler/frame`
+     where `baseUrl` is the URL where your Backstage backend can be reached;
+     note that VMware Cloud does not support the combination of an `http://`
+     scheme and a `localhost` hostname, so when testing locally it may help to
+     set your backend base URL to `http://127.0.0.1:7007`.
+   - `Refresh Token`: check `Issue refresh token`; refresh tokens are required
+     to prevent forcing users to re-login when they refresh their browser.
+   - `Define Scopes`: check `OpenID` at the bottom.
+1. Click **Create**.
+1. Take note of the `App ID` in the resulting modal; this is the client ID to be
+   used by Backstage.
+
+## Install the provider in the backend
+
+### New backend system
+
+Apps using the [new backend system](../../backend-system/index.md),
+can enable the VMware Cloud provider with a small modification like:
+
+```ts title="packages/backend-next/src/index.ts"
+import { createBackend } from '@backstage/backend-defaults';
+
+const backend = createBackend();
+backend.add(import('@backstage/plugin-auth-backend'));
+/* highlight-add-start */
+backend.add(
+  import('@backstage/plugin-auth-backend-module-vmware-cloud-provider'),
```

Comment:
Let's keep this as is for the new system. We're going to move all modules to be separate and not be available as built-in providers of the `auth-backend` at all, so if we added this as a built-in provider in the new system we'd just force another migration in the future.

Reply:
Agreed!